assignment4/disjointset.py

```diff
@@ -0,0 +1,37 @@
+class DisjointSet:
+    """Implementation of a Disjoint Set. Takes integers as keys."""
+
+    def __init__(self):
+        self.parent = list()
+        self.rank = list()
+        self.setAmount = 0
+
+    def getSetAmount(self):
+        return self.setAmount
+
+    def makeSet(self, v):
+        if len(self.parent) <= v:
+            toAdd = v - len(self.parent) + 1
+            self.parent += [0] * toAdd
```

Comment:
The way the API for this class is written, it is possible to call `makeSet` with any arbitrary value `v`.

One alternative would be to have `makeSet` returns the index of the newly create set, instead of taking it as an argument. That way, you always have to add a single new element to the arrays.

In your current usage of the `makeSet` in the code above, you already always call it using consecutive values, so it should not make that code any more complex.